windowing_library.py

```diff
@@ -0,0 +1,71 @@
+import pickle
+import matplotlib.pyplot as plt
+import numpy as np
+import skimage 
+from scipy import signal
+from scipy.spatial import procrustes
+from skimage.util.shape import view_as_windows
+import statistics
+
+# make it usable for all rows and make one function you call first for phase 1 and 2 then a second main function for phase 3
+
+def create_windows(matrix, rows, window_size, stride, mod_num):
+  x = matrix['x']
+  y = matrix['y']
+  z = matrix['z']
+  nanos = matrix['nanos'] 
+  np_data = np.array([x,y,z,nanos], 'd')
+
+  windowed_data = view_as_windows(np_data, (rows, window_size), step=stride)
+  return windowed_data
+
+def find_initial_lift_times(lift_windows, rows):
+  start = lift_windows[0][rows-1][0]
+  end = lift_windows[-1][-1][-1]
+  return start, end
+
+def find_lift_windows(windowed_data):
+  variance=[]
+  lift_windows =[]
+  windows = windowed_data[0]
+  for i in range(len(windows)):
+	x = windows[i][0]
+	y = windows[i][1]
+	z = windows[i][2]
+	nanos = windows[i][3]
+	var_total = statistics.variance(x) + statistics.variance(y) + statistics.variance(z)
+	variance.append(var_total)
+  threshold = max(variance)/4
+  for i in range(len(windows)): 
+     if variance[i] >= threshold:
+       lift_windows.append(windows[i])
+       if (i != len(variance)-1) and (variance[i+1] < threshold):
+        break
+  return lift_windows
+
+def start_time_on_smaller_windows(windowed_data, threshold):
+  start_time =[]
+  windows = windowed_data[0]
+  for i in range(len(windows)):
+    x = windows[i][0]
+    y = windows[i][1]
+    z = windows[i][2]
+    nanos = windows[i][3]
+    var_total = statistics.variance(x) + statistics.variance(y) + statistics.variance(z)
```

Comment:
I've seen that code before. If you use it more than once it's definitely worth putting it into a function ;-)

But if it's only as single np call after updating the above code, it might not be worth it.